From caf19a2f858d7616aa1fdbc9c62e3cf67e42b2d9 Mon Sep 17 00:00:00 2001
From: Patrick Spendrin <ps_ml@gmx.de>
Date: Tue, 3 Apr 2012 07:38:44 +0200
Subject: [PATCH 1/4] fix windows vs10 build

---
 python-src/PC/pyconfig.h |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/python-src/PC/pyconfig.h b/python-src/PC/pyconfig.h
index cbddbd8..61d2916 100644
--- a/python-src/PC/pyconfig.h
+++ b/python-src/PC/pyconfig.h
@@ -191,12 +191,14 @@ WIN32 is still required for the locale module.
 #endif
 
 /* Define like size_t, omitting the "unsigned" */
+#if defined(_MSC_VER) && _MSC_VER <= 1500
 #ifdef MS_WIN64
 typedef __int64 ssize_t;
 #else
 typedef _W64 int ssize_t;
 #endif
 #define HAVE_SSIZE_T 1
+#endif
 
 #if defined(MS_WIN32) && !defined(MS_WIN64)
 #ifdef _M_IX86
-- 
1.7.4.msysgit.0

