From 0ac84a6f945cd76acea837c09128b44b0709b333 Mon Sep 17 00:00:00 2001
From: Patrick Spendrin <ps_ml@gmx.de>
Date: Tue, 3 Apr 2012 07:47:12 +0200
Subject: [PATCH 3/4] fix class ranges with inheritance

---
 python-src/Python/ast.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/python-src/Python/ast.c b/python-src/Python/ast.c
index 776838d..7d52aab 100644
--- a/python-src/Python/ast.c
+++ b/python-src/Python/ast.c
@@ -3050,7 +3050,7 @@ ast_for_classdef(struct compiling *c, const node *n, asdl_seq *decorator_seq)
         if (forbidden_name(classname, CHILD(n, 3), 0))
             return NULL;
         return ClassDef(classname, NULL, NULL, NULL, NULL, s, decorator_seq,
-                        LINENO(n), n->n_col_offset, c->c_arena);
+                        LINENO(CHILD(n, 1)), CHILD(n, 1)->n_col_offset, c->c_arena);
     }
 
     if (TYPE(CHILD(n, 3)) == RPAR) { /* class NAME '(' ')' ':' suite */
@@ -3063,7 +3063,7 @@ ast_for_classdef(struct compiling *c, const node *n, asdl_seq *decorator_seq)
         if (forbidden_name(classname, CHILD(n, 3), 0))
             return NULL;
         return ClassDef(classname, NULL, NULL, NULL, NULL, s, decorator_seq,
-                        LINENO(n), n->n_col_offset, c->c_arena);
+                        LINENO(CHILD(n, 1)), CHILD(n, 1)->n_col_offset, c->c_arena);
     }
 
     /* class NAME '(' arglist ')' ':' suite */
@@ -3074,7 +3074,7 @@ ast_for_classdef(struct compiling *c, const node *n, asdl_seq *decorator_seq)
         dummy_name = NEW_IDENTIFIER(CHILD(n, 1));
         if (!dummy_name)
             return NULL;
-        dummy = Name(dummy_name, Load, LINENO(n), n->n_col_offset, c->c_arena);
+        dummy = Name(dummy_name, Load, LINENO(CHILD(n, 1)), CHILD(n, 1)->n_col_offset, c->c_arena);
         call = ast_for_call(c, CHILD(n, 3), dummy);
         if (!call)
             return NULL;
@@ -3090,7 +3090,7 @@ ast_for_classdef(struct compiling *c, const node *n, asdl_seq *decorator_seq)
 
     return ClassDef(classname, call->v.Call.args, call->v.Call.keywords,
                     call->v.Call.starargs, call->v.Call.kwargs, s,
-                    decorator_seq, LINENO(n), n->n_col_offset, c->c_arena);
+                    decorator_seq, LINENO(CHILD(n, 1)), CHILD(n, 1)->n_col_offset, c->c_arena);
 }
 
 static stmt_ty
-- 
1.7.4.msysgit.0

